src/plugins/intel_cpu/src/transformations/snippets/aarch64/pass/snippets_mark_skipped.cpp

```diff
@@ -0,0 +1,526 @@
+            channelAxis = getChannelAxis(reduce->get_reduction_axes(), reduce->get_keep_dims());
+            SetNodeFusingType(node, NodeFusingType::FusedWithReduce);
+        } else if (isSuitableMiscParent(node)) {
+            if (const auto reduce = std::dynamic_pointer_cast<const ov::op::util::ArithmeticReductionKeepDims>(node)) {
+                channelAxis = getChannelAxis(reduce->get_reduction_axes(), reduce->get_keep_dims());
+            } else {
+                channelAxis = DEFAULT_AXIS;
+            }
+            SetNodeFusingType(node, NodeFusingType::FusedWithMisc);
+        } else if (isSuitableMatMulParent(node)) {
+            const bool is_fc = isFullyConnected(node);
+            const bool is_i8 = canBeMatMulExecutedInInt8(node->get_input_element_type(0), node->get_input_element_type(1));
+            const auto out_rank = node->get_output_partial_shape(0).rank();
+            if (is_fc) {
+                SetNodeFusingType(node, is_i8 ? NodeFusingType::FusedWithFCI8 : NodeFusingType::FusedWithFC);
+                channelAxis = out_rank.is_static() ? (out_rank.get_length() == 3 ? 2 : 1) : DEFAULT_AXIS;
+            } else {
+                SetNodeFusingType(node, is_i8 ? NodeFusingType::FusedWithMatMulI8 : NodeFusingType::FusedWithMatMul);
+                channelAxis = out_rank.is_static() ? out_rank.get_length() - 1 : DEFAULT_AXIS;
+            }
+        } else if (isSuitableSubtractAsZeroPointsParent(node)) {
+            SetSnippetsNodeType(node, snippets::pass::SnippetsNodeType::SkippedByPlugin);
+            channelAxis = DEFAULT_AXIS;
+        // CVS-105447
+        // This WA skip convert with same I/O precision in Snippets
+        // Such useless Convert is executed in Snippets
+        } else if (enableBF16 && isSuitableConvert(node)) {
+            SetSnippetsNodeType(node, snippets::pass::SnippetsNodeType::SkippedByPlugin);
+            channelAxis = DEFAULT_AXIS;
+        } else {
+            for (const auto fusingChainType : getContinuableChains(node)) {
+                if (fusingChainType == NodeFusingType::FusedWithReduce) {
+                    if (isSuitableReduceChild(node, channelAxis))
+                        PropagateIfHasOnlyChild(node, fusingChainType);
+                } else if (isSuitableChildForFusingSimple(node, channelAxis)) {
+                    PropagateIfHasOnlyChild(node, fusingChainType);
+                } else if (fusingChainType == NodeFusingType::FusedWithConvolution ||
+                           fusingChainType == NodeFusingType::FusedWithBinaryConvolution) {
+                    if (isSuitableParentForFusingSumActivation(node)) {
+                        PropagateIfHasOnlyChild(node, NodeFusingType::FusedWithConvolutionSumActivation);
+                        // Mimic FuseConvolutionAndSimpleOperationThroughMaxPool
+                    } else if (isSuitablePoolChild(node)) {
```

Comment:
Thanks Alexandra for the comment! As far as I know, there is no directly way to know if a ngraph conv already fused something. So I plan to add a new flag similar to SetNodeFusingType, to denote if a parent node already fused a child node. Please feel free to discuss, if you have concerns.

Reply:
Yeah, you're right! We don't explicitly know if `Conv` has already fused bias because ngraph op doesn't fusing mechanism. But we know what will be happened in `GraphOptimizer` where there is this mechainsm 😃 

Let me explain how it works on example `Conv -> Bias -> Sum` for x64:
Firstly, we mark `Conv` by `FusedWithConvolution` [here](https://github.com/openvinotoolkit/openvino/blob/2426259eaeed41feabc5d7445863c8aa98c5ecf1/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp#L494).
Secondly, we mark `Bias` by the same `fusingChainType` - `FusedWithConvolution` due to it's simple [fusing](https://github.com/openvinotoolkit/openvino/blob/2426259eaeed41feabc5d7445863c8aa98c5ecf1/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp#L537).
Thirdly, we analyze `Sum` node: it's not [simple fusing](https://github.com/openvinotoolkit/openvino/blob/2426259eaeed41feabc5d7445863c8aa98c5ecf1/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp#L537) since the second branch is not constant. Then we see that the parent of `Sum` - `Bias` has `fusingChainType` -  `FusedWithConvolution`, so we step in [isSuitableParentForFusingSumActivation](https://github.com/openvinotoolkit/openvino/blob/2426259eaeed41feabc5d7445863c8aa98c5ecf1/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp#L541C26-L541C63). This check should return `True` and mark `Sum` as `FusedWithConvolutionSumActivation`.
Thus, all three nodes are [marked](https://github.com/openvinotoolkit/openvino/blob/2426259eaeed41feabc5d7445863c8aa98c5ecf1/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp#L563) as `Skipped`.

The algorithm propagates `fusingChainType` from the node to children and change type to another `fusingChainType` if needed to allow other `isSuitable<>` checks verify the own conditions for fusions. Inother words, this is a emulation of fusing mechanism.

If you have questions, feel free to ask my or @IvanNovoselov as author of this complex but very useful algorithm 😃

Reply:
Applied. Thanks Alexandra!